y.coin-selection/coin-selection.md

```diff
@@ -0,0 +1,57 @@
+After achieving these requirements, other considerations come into play.
+
+# Dimensions of the Problem
+
+## Smallest transaction pays lowest fees
+Generally, the smallest possible transaction is the cheapest to get included in a block.
+So, it may seem obvious that one could save fees by always building transactions with a single input.
+However, this would likely require the creation of a change ouput which would mean that the sending wallet's UTXO count would remain constant, while new UTXO get created for the recipients.
+If the wallet also frequently receives incoming transactions, optimizing for this dimension exclusively can lead to a quickly bloating UTXO pool.
+
+## Every UTXO has to be spent eventually
+Given that Bitcoin balance is only accessible[^1] by spending the corresponding UTXO, every UTXO must be spent eventually.
+Assuming that there will be some periods of time in the future in which block inclusion remains continuously restricted to high fee rates, it may be preferable to spend low-value UTXO at earlier low-fee periods.
+Otherwise, low-value UTXO may become economically unspendable for the duration of such a fee rate spike when the cost to spend a UTXO exceeds the UTXO's value.
+Strategies to pre-empt this situation include consolidating UTXO periodically, using different coin selection strategies at different fee rate levels, or simply prioritizing the consumption of low-value UTXO.
+
+## Privacy
+While Bitcoin has some nice privacy property in the sense that payments are associated only with a pseudonym of the owner rather than their identity immediately, it is far from anonymous.
+Where people are concerned to limit their traceability and improve their privacy, they may have vastly different requirements for their coin selection.
+Chainalysis has generally been doing well to assume that any UTXO used together in one transaction indicate being owned by the same wallet.
+[Coinjoin with 100 participants via Wasabi]
+[Spend everything from same address.]
+[Address reuse.]
+[Manual selection.]
+[Separation of funds by concerns.]
+[Coinjoin with recipient.]
+
+## Individual Usage Pattern
+- Small UTXO Pools for end-users
+- Large incoming to outgoing imbalance for merchants
+- High frequency of sending makes batching viable and attractive
+- Sending frequency correlated with trading activity corresponds to times of high fees (-> consolidations)
+
+## Global UTXO Set
+- CONOP influence of large UTXO Set
+
+## Significance of omitting change outputs
+
+# Simple workable solution: Random Selection
+
+# Heuristic-based UTXO Management, e.g. Branch and Bound
+
+# Coin Control
+
+# Do nots
+## Largest First Selection
+
+# Additional UTXO Management Strategies (link to consolidation and batching)
+# Outlook: New Address formats
+
+[^1]: Barring direct transfer of UTXO ownership e.g. per hardware wallets or layer-two constructions such as state-chains.
```

Comment:
Should we mention needing to refresh timelocks? For example, the Green wallet considers this in their coin selection algorithm.